astrovascpy/bloodflow.py

```diff
@@ -654,15 +652,19 @@ def _solve_linear(laplacian, input_flow):
         if sparse.issparse(input_flow):
             input_flow = input_flow.toarray()
 
-        result = np.zeros(np.shape(laplacian)[0], dtype=laplacian.dtype)
+        n_nodes = np.shape(laplacian)[0]
```

Comment:
Isn't it the same?
```suggestion
        n_nodes = len(laplacian)
```

Reply:
For sparse matrices, the `len` is not defined.  We must use `shape[0]`.